core/src/executor/aggregate/state.rs

```diff
@@ -244,15 +479,27 @@ impl<'a, T: GStore> State<'a, T> {
         filter_context: Option<Arc<RowContext<'a>>>,
         aggr: &'a Aggregate,
     ) -> Result<State<'a, T>> {
-        let value = match aggr {
-            Aggregate::Count(CountArgExpr::Wildcard) => Value::Null,
-            Aggregate::Count(CountArgExpr::Expr(expr))
-            | Aggregate::Sum(expr)
-            | Aggregate::Min(expr)
-            | Aggregate::Max(expr)
-            | Aggregate::Avg(expr)
-            | Aggregate::Variance(expr)
-            | Aggregate::Stdev(expr) => evaluate(self.storage, filter_context, None, expr)
+        let value = match &aggr.func {
+            AggregateFunction::Count(CountArgExpr::Wildcard) => {
+                if aggr.distinct {
+                    // For COUNT(DISTINCT *), we need to use all column values as a composite value
+                    let context = filter_context.as_ref().expect(
+                        "filter_context should always be Some in current GlueSQL architecture",
+                    );
```

Comment:
could we remove to return error instead of expect?

Reply:
@panarch Fixed in 6602e9c2 - replaced `expect()` with `ok_or_else()` and added a new
  `FilterContextRequiredForAggregate` error variant for proper error handling.